p2panda-js/src/instance/instance.ts

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+import { Fields, InstanceRecord } from '~/types';
+import { marshallRequestFields } from '~/utils';
+
+import { getMessageFields } from '~/message';
+import { signPublishEntry } from '~/entry';
+import { materializeEntries } from './materialiser';
+
+import type { Context } from '~/session';
```

Comment:
These are really small nitpicks:

* Since we group the imports via newlines I wonder what the system is behind it? All of these import paths seem to belong to the same level and I would suspect them to be in one group, except of the `type` import.
* In terms of consistency it can be cool to stick to either `~` or `./` for importing our own modules and avoid having a mix. I suggest sticking to `~` whereever we import in the codebase as it allows simple copy / paste between documents without caring about the folder structure.

Btw, this is cool: https://github.com/import-js/eslint-plugin-import

Reply:
> Since we group the imports via newlines I wonder what the system is behind it? All of these import paths seem to belong to the same level and I would suspect them to be in one group, except of the type import.

Good point. I also like setting up eslint to manage this, maybe in a separate PR (didn't we already do that somewhere? 🤔)

> I suggest sticking to ~ whereever we import in the codebase as it allows simple copy / paste between documents without caring about the folder structure.

That's an interesting thought. I can see how that's useful when you are copying imports. 

I like using both and choosing whatever results in a shorter import path. Usually that means, importing with `./` when things are in the same directory and using `~/ when they are elsewhere. About this I also like making an explicit distinction between imports from the same directory and imports from elsewhere in the project.

Reply:
@adzialocha please have a look at the new grouping in `instance.ts`. Is this clearer to you?

Reply:
Yes, thank you!!

Reply:
> About this I also like making an explicit distinction between imports from the same directory and imports from elsewhere in the project.

This also sounds good :+1: